pyop2/codegen/rep2loopy.py

```diff
@@ -85,6 +91,210 @@ def petsc_function_lookup(target, identifier):
     return None
 
 
+class INVCallable(loopy.ScalarCallable):
+    def __init__(self, name, arg_id_to_dtype=None,
+                 arg_id_to_descr=None, name_in_target=None):
+
+        super(INVCallable, self).__init__(name,
+                                          arg_id_to_dtype=arg_id_to_dtype,
+                                          arg_id_to_descr=arg_id_to_descr)
+
+        self.name = name
+        self.name_in_target = name_in_target
+
+    def with_types(self, arg_id_to_dtype, kernel, callables_table):
+        for i in range(len(arg_id_to_dtype)):
+            if arg_id_to_dtype.get(i) is None:
+                # the types provided aren't mature enough to specialize the
+                # callable
+                return (self.copy(arg_id_to_dtype=arg_id_to_dtype),
+                        callables_table)
+
+        mat_dtype = arg_id_to_dtype[0].numpy_dtype
+        name_in_target = "inverse"
+
+        return (self.copy(name_in_target=name_in_target,
+                          arg_id_to_dtype={-1: NumpyType(mat_dtype, target=loopy.CTarget()), 0: NumpyType(mat_dtype, target=loopy.CTarget()), 1: NumpyType(numpy.int32, target=loopy.CTarget())}),
```

Comment:
Does one need the target here? The fact that it's not available via `self` or arguments makes me think not.

Reply:
If one does, prefer making a single target, rather than multiple ones of the same type. Also, please wrap the line a bit sooner:

```
{-1: ...,
 0: ..., 
 ...}
```

Reply:
I introduced all the constructs related to Target, when I came up with #580. I think I accidentally mangled them a bit, because I needed both for Tsslac at some point. I will drop the Targets here now, but might need them later.